R/get_gambl_metadata.R

```diff
@@ -563,10 +564,11 @@ get_gambl_metadata = function(seq_type_filter = "genome",
   # define which seq type is not priority
   seq_type_non_priotiry <- ifelse(seq_type_priority == "genome", "capture", "genome")
   # take only those lines that seq_type is not `seq_type_priority` (those were already solved)
-  all_meta <- dplyr::filter(all_meta, ! sample_id %in% samples_with_priority_seq_type)
+  all_meta <- dplyr::filter(all_meta, ! biopsy_id %in% samples_with_priority_seq_type)
```

Comment:
Related to my comment above, it would be better to do:  
```
all_meta <- all_meta %>% 
    group_by(tissue_status, patient_id, biopsy_id) %>% 
    # When > 1 seq_type exists, take the specified priority seq_type
    slice_max(seq_type == seq_type_priority, n=1, with_ties = TRUE) %>% 
    # When still duplicates after this, take the line with the least NAs
    slice_min(na_count, n=1, with_ties = FALSE) %>% 
    ungroup()
```
This takes care of all of this in a single function, I think?

Reply:
My understanding is that these were separated because we always want the mRNA row when it's requested and we only want to prioritize the two DNA seq_types. Would the code you suggested work that way? i.e would we get back mrna + genome (when both genome and capture available)?

Reply:
Hm, I see. This code block is hard to follow because it yo-yos between `all_meta` and `all_meta_res` and those objects aren't named for their contents. I'm really just looking for a solution that uses patient_id and biopsy_id to handle the normals with NA biopsy_id values, because I think that will break here